src/js/components/AppListComponent.jsx

```diff
@@ -186,7 +198,11 @@ var AppListComponent = React.createClass({
 
     if (filterText != null && filterText !== "") {
       nodesSequence = nodesSequence
-        .filter(app => app.id.indexOf(filterText) !== -1);
+        .filter(app => {
+          app.filterScore = score(app.id, filterText);
```

Comment:
Sorry, the links goes to the wrong view type I think. May bad.
But doesn't change the principle how it works.

An additional sort function should be added here: 

https://github.com/mesosphere/marathon-ui/blob/master/src/js/components/AppListComponent.jsx#L301